server/upackages.pas

```diff
@@ -282,13 +276,21 @@ procedure LoadPackageOrProject(const FileName: string);
         Exit;
 
       if UpperCase(ExtractFileExt(FileName)) = '.LPK' then
-        Root := Root.FindNode('Package');
+      begin
+        CompilerOptions := Root.FindNode('Package').FindNode('CompilerOptions');
```

Comment:
This will crash if the XML is malformed and lacks a `<Package>` node. Please add a check for `nil` before dereferencing the node.

Reply:
Done. I created a local function to perform this, and used the same function to item below.